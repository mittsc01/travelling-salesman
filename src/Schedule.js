import React, {useState,useEffect} from 'react'
import {Link} from 'react-router-dom'


export default function Schedule(props){
    const fakeData = [
        {
            id: 0,
            route: {
                id: 1,
                name: 'Test'
            },
            date: "2021-11-13"
        },
        {
            id: 1,
            route: {
                id: 2,
                name: 'Test2'
            },
            date: "2021-11-15"
        },

    ]
    const [runs,setRuns] = useState(fakeData)
    return (
        <div>
            <ul>
        {runs.map(item=><li key={`run-${item.id}`}>
            <Link to={`/schedule/${item.id}`}>{`${item.route.name}`}</Link>
            <button>Edit</button>
            <button>Delete</button>
            </li>)}
    </ul>
    <button onClick={() => props.history.push('/schedule/add')}>Add to Schedule</button>

        </div>
    
    )

}