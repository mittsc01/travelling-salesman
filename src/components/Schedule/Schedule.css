li {
    justify-content: space-around;
}