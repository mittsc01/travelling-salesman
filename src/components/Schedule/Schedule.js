import React, {useState,useEffect} from 'react'
import {Link} from 'react-router-dom'
import RoutesService from '../../services/routes-service'


export default function Schedule(props){
    
    const [runs,setRuns] = useState([])
    useEffect( () => {
        
        (async () => {
            const myRuns = await RoutesService.getSchedule()
            
            
             
    
            
        setRuns(myRuns)})()
    }, [props])

    const handleDelete = async (e,id) => {
        await RoutesService.deleteScheduleItem(id)
        const updatedRuns = await RoutesService.getSchedule()
        setRuns(updatedRuns)
    }
    return (
        <div>
            <ul>
        {runs.map(item=><li key={`run-${item.id}`}>
            <Link to={`/schedule/${item.id}`}>{`${item.title}`}</Link>
            <button onClick={(e) => handleDelete(e,item.id)}>Delete</button>
            </li>)}
    </ul>
    <button onClick={() => props.history.push('/add-to-schedule')}>Add to Schedule</button>

        </div>
    
    )

}