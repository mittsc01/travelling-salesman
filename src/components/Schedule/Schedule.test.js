import Schedule from './Schedule';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Schedule />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Schedule />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });