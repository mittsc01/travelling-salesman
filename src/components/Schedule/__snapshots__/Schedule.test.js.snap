// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <ul
    className="route-list"
  />
  <button
    onClick={[Function]}
  >
    Add to Schedule
  </button>
</div>
`;
